src/main/java/cat/udl/eps/softarch/demo/handler/DonationEventHandler.java

```diff
@@ -17,15 +22,29 @@ public class DonationEventHandler {
     final Logger logger = LoggerFactory.getLogger(User.class);
     final UserRepository userRepository;
     final DonationRepository donationRepository;
+    final RequestRepository requestRepository;
+    final TakeRepository takeRepository;
 
-    public DonationEventHandler(UserRepository userRepository, DonationRepository donationRepository) {
+    public DonationEventHandler(UserRepository userRepository, DonationRepository donationRepository, RequestRepository requestRepository, TakeRepository takeRepository) {
         this.userRepository = userRepository;
         this.donationRepository = donationRepository;
+        this.takeRepository = takeRepository;
+        this.requestRepository = requestRepository;
     }
 
     @HandleBeforeCreate
     public void handleDonationPreCreate(Donation donation) {
         logger.info("Before creating: {}", donation.toString());
+        Request request = requestRepository.findRequestByAmountAndWeight(donation.getAmount(), donation.getWeight());
```

Comment:
It cannot be assumed that a new Donation would be for an existing Request just because it has the same Seed amount and weight. It should be made explicit by the Donor when making the Donation. For instance, in the list of Requests, the Donor selects one and in the details clicks the "Donate" button to create a new Donation for that Request, which will also imply that a Take is generated automatically for the Donation and the Request being fulfilled

Reply:
Both front and backend. First, from the front, the donor is who decides if the Donation is open for takers or targeted to a preexisting Request. 

Unfortunately, it is not possible right now to create a Donation pointing to a Request. To simplify things, the easiest might be to add a new relationship `Donation --targets--> Request` in Donation that point to 0 or 1 Request. The handler in the backend can use that information to create the Donation together with the Take that fulfills the Request

Reply:
Done!